src/domain/entities/Indicator.ts

```diff
@@ -0,0 +1,168 @@
+import _ from "lodash";
+import { Config } from "../../models/Config";
+import { DataElement } from "./DataElement";
+import { Struct } from "./generic/Struct";
+import { IndicatorCategory, IndicatorType } from "./IndicatorType";
+import { Code, Id, NamedRef } from "./Ref";
+
+interface IndicatorAttrs extends NamedRef {
+    code: Code;
+    denominator: IndicatorFormula;
+    numerator: IndicatorFormula;
+    shortName: string;
+    type: IndicatorType;
+    groupName: string;
+}
+
+const COST_BENEFIT_NAME = "Cost / Benefit";
+const TARGET_ACTUAL_NAME = "Actual / Target";
+
+export class Indicator extends Struct<IndicatorAttrs>() {
+    static buildIndicatorsByType(
+        actualIndicatorId: Id,
+        costBenefitIndicatorId: Id,
+        dataElement: IndicatorDataElement,
+        config: Config,
+        allIndicatorsTypes: IndicatorType[]
+    ): Indicator[] {
+        const isPeopleType = dataElement.mainType.name === "people";
+
+        const actualIndicator = this.create({
+            id: actualIndicatorId,
+            name: this.generateName(dataElement, TARGET_ACTUAL_NAME, "actual", allIndicatorsTypes),
+            code: `${config.base.indicators.actualTargetPrefix}${dataElement.code}`,
+            denominator: this.buildActualIndicatorFormula(
+                dataElement,
+                "denominator",
+                config.categoryOptionCombos.target.id
+            ),
+            groupName: "People Target / Actual",
+            numerator: this.buildActualIndicatorFormula(
+                dataElement,
+                "numerator",
+                config.categoryOptionCombos.actual.id
+            ),
+            shortName: this.generateShortName(
+                dataElement,
+                TARGET_ACTUAL_NAME,
+                "actual",
+                allIndicatorsTypes
+            ),
+            type: this.generateIndicatorType(allIndicatorsTypes, "actual"),
+        });
+
+        const benefitIndicator =
+            dataElement.pairedPeople && !isPeopleType
```

Comment:
[minor] as we're building the benefit indicator, it would be more declarative to have `&& isBenefitType`. This would also work nice if we have more types in the future.